### PR TITLE
CCN-lite reimport

### PR DESCRIPTION
**Pull request by OlegHahm**
~~This is WIP for re-integrating ccn-lite as a package in RIOT using either sockets or netapi.~~

~~Waiting for #4358 and #4429.~~

~~Depends on #4398.~~

### CODE REVIEW

File: pkg/ccn-lite/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	cp $(PKG_DIR)/src/lib-ccn-lite.a ${BINDIR}/ccn-lite.a
+	cp $(PKG_DIR)/src/lib-ccn-lite-utils.a ${BINDIR}/ccn-lite-utils.a
+
+$(PKG_DIR)/Makefile: $(PKG_DIR)/.git/config
+
+$(PKG_DIR)/.git/config:
+	test -d "$(PKG_DIR)" || git clone "$(PKG_URL)" "$(PKG_DIR)"; \
+		cd "$(PKG_DIR)" && git checkout -f "$(PKG_VERSION)"
+
+clean::
```
Comment (cgundogan) [inline pkg/ccn-lite/Makefile]: question: why double `:` here and for `distclean`?
I noticed that this is also the case for the `libcoap`'s and `oonf_api`'s Makefile.

Comment (gebart) [inline pkg/ccn-lite/Makefile]: http://stackoverflow.com/questions/7891097/what-are-double-colon-rules-in-a-makefile-for

Comment (cgundogan) [inline pkg/ccn-lite/Makefile]: thanks!

Comment (OlegHahm) [inline pkg/ccn-lite/Makefile]: To be honest, was mostly copy&paste here.

**Comment by OlegHahm** [comment]
This is ready for testing.

**Comment by emmanuelsearch** [comment]
Tested on native, with NDN format, 1 node sending the interest, and the other serving the content.
Works!

Please add README, and change the shell common names to add a self-explanatory prefix such as ccn-*

**Comment by emmanuelsearch** [comment]
README looks good. How about also adding something about using ifconfig to obtain the id to feed to ccnl-open?

**Comment by OlegHahm** [comment]
updated README

**Comment by OlegHahm** [comment]
Hm, CCN-Lite uses a lot of malloc. I tested with TLSF and it seems to work - should I add this to the PR?

**Comment by emmanuelsearch** [comment]
What do you mean exactly with "add this to the PR"?

**Comment by emmanuelsearch** [comment]
no strong opinion on this. I guess that if CCN-lite uses a lot of dynamic memory, we have to find a solution anyways, so why not this one?

**Comment by OlegHahm** [comment]
I think the problem without tlsf is that the current implementation of CCN-Lite doesn't check the return value for `malloc` in a lot of (all?) places, which is particular problematic if `free` is not really working on some (most of our) platforms. The version of `free` in tlsf should work and hence, drastically decrease the chances of a hardfault.

**Comment by emmanuelsearch** [comment]
OK, so let's try TLSF (unless someone objects).

**Comment by OlegHahm** [comment]
did

**Comment by OlegHahm** [comment]
Are you okay with squashing?

**Comment by emmanuelsearch** [comment]
Yes, please squash.

**Comment by OlegHahm** [comment]
squashed

**Comment by OlegHahm** [comment]
All dependencies are merged, let's wait for Travis.

**Comment by OlegHahm** [comment]
Yay!

**Comment by authmillenon** [comment]
Any reason why this was preferred over #4289?

**Comment by authmillenon** [comment]
Ah, did not see, that this updated CCN-lite, just that it imported a `pkg/doc.txt`

**Comment by authmillenon** [comment]
(would rather prefer the proposed doc in #4289 over the stub imported here anyways ;-))

**Comment by OlegHahm** [comment]
Sorry about that. I agree it was a bit sneaky to introduce this change here, but it didn't want to outsource to yet another PR that this one would have been dependent on. I think we can improve documentation and its structure during the next days.

**Comment by kaspar030** [comment]
btw, YAY! ;)

**Pull request merged**